### ISSUE TITLE
Add `axis` argument to `lax.top_k`

### ISSUE THREAD

**Issue by carlosgmartin** [jax-ml/jax#25867]
**Feature request:** Add an `axis` argument to [`lax.top_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.top_k.html).

Other similar reduction functions in [`lax`](https://jax.readthedocs.io/en/latest/jax.lax.html) receive an `axis` or `reduction_dimension` argument:
- [`argmax`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.argmax.html)
- [`argmin`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.argmin.html)
- [`approx_max_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.approx_max_k.html)
- [`approx_min_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.approx_min_k.html)

**Comment by aymuos15** [comment]
Not sure if still relevant but could i take this up?

**Comment by carlosgmartin** [comment]
@aymuos15 Go for it.

**Comment by jakevdp** [comment]
Just a note here: functions in `jax.lax` are typically just python wrappers around direct calls to primitives (e.g. `lax.top_k` just binds `lax.top_k_p`) and so any solution here would involve adding an `axis` argument to the `top_k_p` primitive, not just to the `top_k` function. This is a bit tricky because the top k primitive defined in `chlo`/StableHLO does not support an axis argument, and the appropriate transpositions would have to also be added to the abstract evaluation rule, batching rule, jvp rule, etc.

**Comment by jakevdp** [comment]
I sent a PR implementing this in #33116
